vm/devices/tpm/src/lib.rs

```diff
@@ -795,24 +799,27 @@ impl Tpm {
 
         tracing::trace!("Request AK cert renewal");
 
-        // Create a new report if `ak_cert_type` is `HwAttested`.
-        let attestation_report = if let TpmAkCertType::HwAttested(_, _) = self.ak_cert_type {
-            Some(self.renew_attestation_report()?)
-        } else {
-            None
-        };
+        let ak_cert_request = self.create_ak_cert_request()?;
+        // Store the ak cert request that includes the attestation report if `ak_cert_type` is `HwAttested`.
+        if matches!(self.ak_cert_type, TpmAkCertType::HwAttested(_)) {
+            self.renew_attestation_report(&ak_cert_request)?;
+        }
 
-        let request_ak_cert = if let TpmAkCertType::HwAttested(_, callback) = &self.ak_cert_type {
-            callback
-        } else if let TpmAkCertType::Trusted(callback) = &self.ak_cert_type {
-            callback
+        let request_ak_cert_helper = if let TpmAkCertType::HwAttested(helper) = &self.ak_cert_type {
```

Comment:
I suggest packaging up this common pattern that you have in a few places now into a `self.ak_cert_type.get_helper() -> Option<Box<dyn ...>>` method